src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -119,18 +121,30 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
         data class VariantInfo(val variant: RsEnumVariant, val alreadyReported: Boolean)
 
         var discrCounter = 0L
+        val reprType = (o.parent as? RsEnumItem)?.reprType ?: return
         val indexToVariantMap = hashMapOf<Long, VariantInfo>()
         for (variant in o.enumVariantList) {
-            val literal = variant.variantDiscriminant?.expr as? RsLitExpr
-            val int = literal?.integerValue
+            val expr = variant.variantDiscriminant?.expr
+            val int = if (expr != null) {
+                val result = RsConstExprEvaluator.evaluate(expr, reprType) as? ExprValue.Integer
+                result?.value ?: return
+            } else {
+                null
+            }
             val idx = int ?: discrCounter
             discrCounter = idx + 1
 
             val previous = indexToVariantMap[idx]
             if (previous != null) {
-                RsDiagnostic.DuplicateEnumDiscriminant(variant, idx).addToHolder(holder)
+                RsDiagnostic.DuplicateEnumDiscriminant(
+                    variant.variantDiscriminant ?: variant,
```

Comment:
I think we want to use expression from `variant.variantDiscriminant` not to highlight `=`

Reply:
Fixed.